pkg/codegen/testing/test/testdata/azure-native-nested-types/java/src/main/java/com/pulumi/azurenative/documentdb/outputs/CompositePathResponse.java

```diff
@@ -52,30 +45,32 @@ public static Builder builder() {
     public static Builder builder(CompositePathResponse defaults) {
         return new Builder(defaults);
     }
-
+    @CustomType.Builder
     public static final class Builder {
         private @Nullable String order;
         private @Nullable String path;
-
-        public Builder() {
-    	      // Empty
-        }
-
+        public Builder() {}
         public Builder(CompositePathResponse defaults) {
     	      Objects.requireNonNull(defaults);
     	      this.order = defaults.order;
     	      this.path = defaults.path;
         }
 
+        @CustomType.Setter
         public Builder order(@Nullable String order) {
             this.order = order;
             return this;
         }
+        @CustomType.Setter
         public Builder path(@Nullable String path) {
             this.path = path;
             return this;
-        }        public CompositePathResponse build() {
-            return new CompositePathResponse(order, path);
+        }
+        public CompositePathResponse build() {
```

Comment:
Looking good! So this removed the situation with "too many constructors". 

Observing that since class fields are no longer final, it's possible to shorten the builder code further by keeping an instance of the class.

```
    @CustomType.Builder
    public static final class Builder {
        private @Nullable CompositePathResponse $;
    }
    @CustomType.Setter
    public Builder path(@Nullable String path) {
         $.path = path;
         return this;
    }
    public CompositePathResponse build() {
        return $;
    }
```

Reply:
yes, correct, I've noticed it too, just didn't want to make unrelated changes